src/index.tsx

```diff
@@ -137,14 +137,7 @@ export default class InfiniteScroll extends Component<Props, State> {
     }
   }
 
-  UNSAFE_componentWillReceiveProps(props: Props) {
-    // do nothing when dataLength and key are unchanged
-    if (
-      this.props.key === props.key &&
-      this.props.dataLength === props.dataLength
-    )
-      return;
```

Comment:
Return here. you are enforcing that data must be changed every time after a refresh call.

Reply:
Right, I see now. How to determine the willReceiveProps trigger is when the refresh call returned.

If the parent component re-rendered because of it's on change, the willReceiveProps will falsely mark actionTriggered = false.